sound/soc/sof/ipc4-pcm.c

```diff
@@ -561,14 +561,14 @@ static int sof_ipc4_get_stream_start_offset(struct snd_sof_dev *sdev,
 	struct sof_ipc4_pipeline_registers ppl_reg;
 	u64 stream_start_position;
 	u32 dai_sample_size;
-	u32 ch, node_id;
+	u32 ch, node_index;
 	u32 offset;
 
-	node_id = SOF_IPC4_NODE_INDEX(host_copier->data.gtw_cfg.node_id);
-	if (node_id == SOF_IPC4_INVALID_NODE_ID)
+	node_index = SOF_IPC4_NODE_INDEX(host_copier->data.gtw_cfg.node_id);
+	if (host_copier->data.gtw_cfg.node_id == SOF_IPC4_INVALID_NODE_ID)
 		return -EINVAL;
```

Comment:
it would be better to swap these to be easier to follow:
```c
if (host_copier->data.gtw_cfg.node_id == SOF_IPC4_INVALID_NODE_ID)
		return -EINVAL;

node_index = SOF_IPC4_NODE_INDEX(host_copier->data.gtw_cfg.node_id);
```

Reply:
I did it first, but I found it resulted to great change in diff, so do it this way.  Let's do it in

Reply:
@RanderWang, thank you!